README.md

```diff
@@ -1,16 +1,248 @@
+
+Once you open a method definition with the `def` keyword, all subsequent lines
+in your program are considered the method's body, the actual procedure or code
+that your method will run every time it's called.
+
+You must terminate every opening `def` of a method with a corresponding `end` in
+order to close the method body. If you don't correctly `end` a method, your
+program will have unexpected results or break entirely because of a syntax
+error. A good practice is to define the method and then immediately close it
+before programming anything into the method.
+
+```ruby
+def greeting
+  # Leave a line break for the method body
+end # Immediately close the method.
+```
+
+Here we set up the method's structure first, ensuring a proper termination
+before adding any other complexity. It's also a great practice to indent methods
+correctly. The body of a method should be indented two (2) spaces, placing it
+visually within the method. When you `end` the method, go back to the same
+indentation of the `def`, aligning the opening and closing of the method
+visually.
+
+Then you can easily define the body of the method and never worry about
+forgetting to `end` the method.
+
+```ruby
+def greeting
+  puts "Hello World" # Now code the body of the method.
+end
+```
+
+## Recognize How to Call Methods
+
+Once you define a method, you can execute the method whenever you want by using
+the method name in your code. Methods should be defined before calling them,
+otherwise Ruby will raise an exception for undefined method invoking.
+
+```ruby
+def greeting
+  puts "Hello World"
+end
+
+greeting # Executing the method by name
+#> "Hello World"
+
+greeting # Executing the method again
+#> "Hello World"
+```
+
+Try it out. Make a new file called `greeting.rb` (you can use: `touch
+greeting.rb` from your terminal). Put the following code in it:
+
+File: `greeting.rb`
+
+```ruby
+def greeting
+  puts "Hello World"
+end
+```
+
+Save your file and run it with `ruby greeting.rb`. You'll see:
+
+```bash
+$ ruby greeting.rb
+$
+```
+
+You'll notice that when you run your program, nothing happens. Your program
+successfully defined the method but it never executed it. Just because you built
+a machine doesn't mean that you turned it on. Update your `greeting.rb` to
+entirely read:
+
+File: `greeting.rb`
+
+```ruby
+def greeting
+  puts "Hello World"
+end
+
+greeting
+```
+
+Save your file and run it with `ruby greeting.rb`. You'll see:
+
+```bash
+$ ruby greeting.rb
+Hello World
+$
+```
+
+Now your program actually executed the program. Update the code again to
+entirely read:
+
+File: `greeting.rb`
+
+```ruby
+def greeting
+  puts "Hello World"
+end
+
+greeting
+greeting
+greeting
+greeting
+greeting
+```
+
+Save your file and run it with `ruby greeting.rb`. You'll see:
+
+```bash
+$ ruby greeting.rb
+Hello World
+Hello World
+Hello World
+Hello World
+Hello World
+$
+```
+
+The bareword `greeting` will execute the body of the defined method.
+
```

Comment:
lets make this an LG and keep it. I want us to be safe to say `Object#razzle` safely and KNOW that we taught it somewhere.